### PR TITLE
que5.3

### PR DESCRIPTION
**Pull request by ashok-choudhary-0**

### CODE REVIEW

File: config/firebaseConfig.js [hunk 1]
```diff
@@ -0,0 +1,14 @@
+const { initializeApp } = require("firebase/app");
```
Comment (amrit56789) [inline config/firebaseConfig.js]: new text document.txt ?

File: config/firebaseConfig.js [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+const { getAuth } = require("firebase/auth");
+const firebaseConfig = {
```
Comment (amrit56789) [inline config/firebaseConfig.js]: instead of this you can use serviceAccount.json file.

File: controllers/authController.js [hunk 3]
```diff
@@ -0,0 +1,16 @@
+const { createUserWithEmailAndPassword } = require("firebase/auth")
```
Comment (amrit56789) [inline controllers/authController.js]: not auth controller you are creating  user.

File: index.js [hunk 4]
```diff
@@ -0,0 +1,13 @@
+const express = require("express")
+const bodyParser = require("body-parser")
+require("dotenv").config();
+const authRoutes = require("./routes/authRoutes")
```
Comment (amrit56789) [inline index.js]: not auth its users.

File: index.js [hunk 5]
```diff
@@ -0,0 +1,13 @@
...
+const port = process.env.port || 5000;
+
+const app = express();
+app.use(bodyParser.json())
+app.use("/auth", authRoutes)
```
Comment (amrit56789) [inline index.js]: /user

File: controllers/authController.js [hunk 6]
```diff
@@ -0,0 +1,16 @@
...
+const auth = require("../config/firebaseConfig")
+
+const signup = async (req, res) => {
```
Comment (amrit56789) [inline controllers/authController.js]: createUser

File: controllers/authController.js [hunk 7]
```diff
@@ -0,0 +1,16 @@
...
+  const { email, password } = req.body;
+  if (!email || !password) {
+    res.status(404).send({ message: "Please fill all the required fields" })
+  }
+  try {
+    const data = await createUserWithEmailAndPassword(auth, email, password);
```
Comment (amrit56789) [inline controllers/authController.js]: newUser.

File: .gitignore [hunk 8]
```diff
@@ -0,0 +1,3 @@
+node_modules
+.env
+package-lock.json
```
Comment (meanstack9716) [inline .gitignore]: Package lock should not be here

File: config/firebaseConfig.js [hunk 9]
```diff
@@ -0,0 +1,20 @@
...
+  projectId: process.env.projectId,
+  storageBucket: process.env.storageBucket,
+  messagingSenderId: process.env.messagingSenderId,
+  appId: process.env.appId,
+  measurementId: process.env.measurementId
+};
+const app = initializeApp(firebaseConfig);
+const auth = getAuth(app);
+module.exports = { auth };
+
```
Comment (meanstack9716) [inline config/firebaseConfig.js]: Why too many white slaces?

**Review by meanstack9716 — CHANGES_REQUESTED** [review]

File: controllers/userController.js [hunk 10]
```diff
@@ -0,0 +1,17 @@
+const admin = require('firebase-admin');
+const signup = async (req, res) => {
+  const {email, password} = req.body;
```
Comment (yuvrajInno) [inline controllers/userController.js]: code not formatted @ashok-choudhary-0

File: controllers/userController.js [hunk 11]
```diff
@@ -0,0 +1,17 @@
+const admin = require('firebase-admin');
+const signup = async (req, res) => {
```
Comment (yuvrajInno) [inline controllers/userController.js]: function name should be camleCase @ashok-choudhary-0

**Review by yuvrajInno — CHANGES_REQUESTED** [review]

**Pull request merged**